experimental/packages/opentelemetry-sdk-metrics-base/src/view/View.ts

```diff
@@ -14,53 +14,107 @@
+import { InstrumentType } from '../InstrumentDescriptor';
 
-// https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/sdk.md#view
+export type ViewOptions = {
+  // View
+  /**
+   *  If not provided, the Instrument name will be used by default. This will be used as the name of the metrics stream.
+   */
+  name?: string,
+  /**
+   * If not provided, the Instrument description will be used by default.
+   */
+  description?: string,
+  /**
+   * If provided, the attributes that are not in the list will be ignored.
+   * If not provided, all the attribute keys will be used by default.
+   */
+  attributeKeys?: string[],
+  /**
+   * The {@link Aggregation} aggregation to be used.
+   */
+  aggregation?: Aggregation,
 
-export interface ViewStreamConfig {
+  // Instrument
+  /**
+   * The type of the Instrument(s).
+   */
+  instrumentType?: InstrumentType,
   /**
-   * the name of the resulting metric to generate, or null if the same as the instrument.
+   * Name of the Instrument(s) with wildcard support.
    */
-  name?: string;
+  instrumentName?: string,
+
+  // Meter
   /**
-   * the name of the resulting metric to generate, or null if the same as the instrument.
+   * The name of the Meter.
    */
-  description?: string;
+  meterName?: string;
   /**
-   * the aggregation used for this view.
+   * The version of the Meter.
    */
-  aggregation?: Aggregation;
+  meterVersion?: string;
   /**
-   * processor of attributes before performing aggregation.
+   * The schema URL of the Meter.
    */
-  attributesProcessor?: AttributesProcessor;
+  meterSchemaUrl?: string;
+};
+
+function isSelectorProvided(options: ViewOptions): boolean {
+  return (options.instrumentName == null &&
+    options.instrumentType == null &&
+    options.meterName == null &&
+    options.meterVersion == null &&
+    options.meterSchemaUrl == null);
 }
 
-/**
- * A View provides the flexibility to customize the metrics that are output by
- * the SDK. For example, the view can
- * - customize which Instruments are to be processed/ignored.
- * - customize the aggregation.
- * - customize which attribute(s) are to be reported as metrics dimension(s).
- * - add additional dimension(s) from the {@link Context}.
- */
 export class View {
   readonly name?: string;
   readonly description?: string;
   readonly aggregation: Aggregation;
   readonly attributesProcessor: AttributesProcessor;
+  readonly instrumentSelector: InstrumentSelector;
+  readonly meterSelector: MeterSelector;
 
-  /**
-   * Construct a metric view
-   *
-   * @param config how the result metric streams were configured
-   */
-  constructor(config?: ViewStreamConfig) {
-    this.name = config?.name;
-    this.description = config?.description;
-    this.aggregation = config?.aggregation ?? Aggregation.Default();
-    this.attributesProcessor = config?.attributesProcessor ?? AttributesProcessor.Noop();
+  constructor(viewOptions: ViewOptions) {
```

Comment:
I find that `isViewOptionsEmpty` is no longer applied on creating the views. Is there any specific reason for removing that?

Reply:
When I implemented `isViewOptionsEmpty()` in #2820 I read the spec wrong, and it is actually not disallowed to create an empty view, but it is disallowed to create view without selector options specified. I explain it in more detail and link to the spec section in question in this comment [here](https://github.com/open-telemetry/opentelemetry-js/pull/3066#discussion_r916616400).